Code/DistGeom/DistGeomUtils.cpp

```diff
@@ -466,4 +467,55 @@ ForceFields::ForceField *constructPlain3DForceField(
 
   return field;
 }  // constructPlain3DForceField
+
+ForceFields::ForceField *construct3DImproperForceField(
+    const BoundsMatrix &mmat, RDGeom::Point3DPtrVect &positions,
+    const std::vector<std::vector<int> > &improperAtoms,
+    const std::vector<int> &atomNums) {
+  (void)atomNums;
+  unsigned int N = mmat.numRows();
+  CHECK_INVARIANT(N == positions.size(), "");
+  ForceFields::ForceField *field =
+      new ForceFields::ForceField(positions[0]->dimension());
+  for (unsigned int i = 0; i < N; ++i) {
+    field->positions().push_back(positions[i]);
+  }
+
+  // improper torsions / out-of-plane bend / inversion
+  double oobForceScalingFactor = 10.0;
+  for (unsigned int t = 0; t < improperAtoms.size(); ++t) {
```

Comment:
I don’t think this takes much time. The potentially slow part is the minimization step.


On 28 Jun 2016, at 21:25, Greg Landrum <notifications@github.com> wrote:

> In Code/DistGeom/DistGeomUtils.cpp:
> 
> > +ForceFields::ForceField *construct3DImproperForceField(
> > +    const BoundsMatrix &mmat, RDGeom::Point3DPtrVect &positions,
> > +    const std::vector<std::vector<int> > &improperAtoms,
> > +    const std::vector<int> &atomNums) {
> > +  (void)atomNums;
> > +  unsigned int N = mmat.numRows();
> > +  CHECK_INVARIANT(N == positions.size(), "");
> > +  ForceFields::ForceField *field =
> > +      new ForceFields::ForceField(positions[0]->dimension());
> > +  for (unsigned int i = 0; i < N; ++i) {
> > +    field->positions().push_back(positions[i]);
> > +  }
> > +
> > +  // improper torsions / out-of-plane bend / inversion
> > +  double oobForceScalingFactor = 10.0;
> > +  for (unsigned int t = 0; t < improperAtoms.size(); ++t) {
> I wonder if the time spent here justifies the added complexity.
> @sriniker do you have a sense for how much time is spent in this part of the code?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>